src/currencyEngineXMR.js

```diff
@@ -190,44 +189,26 @@ class MoneroEngine {
   // ***************************************************
   async checkAddressInnerLoop () {
     try {
-      const result = await this.fetchPostMyMonero('get_address_info')
-      const valid = validateObject(result, {
-        'type': 'object',
-        'properties': {
-          'locked_funds': {'type': 'string'},
-          'total_received': {'type': 'string'},
-          'total_sent': {'type': 'string'},
-          'scanned_height': {'type': 'number'},
-          'scanned_block_height': {'type': 'number'},
-          'start_height': {'type': 'number'},
-          'transaction_height': {'type': 'number'},
-          'blockchain_height': {'type': 'number'}
-        }
-      })
+      const params: QueryParams = {
+        moneroAddress: this.walletLocalData.moneroAddress,
+        moneroSpendKeyPrivate: this.walletInfo.keys.moneroSpendKeyPrivate,
+        moneroSpendKeyPublic: this.walletInfo.keys.moneroSpendKeyPublic,
+        moneroViewKeyPrivate: this.walletLocalData.moneroViewKeyPrivate
+      }
 
-      if (valid) {
-        const parsedAddrInfo = moneroResponseParserUtils.Parsed_AddressInfo__sync(
-          this.keyImageCache,
-          result,
-          this.walletLocalData.moneroAddress,
-          this.walletLocalData.moneroViewKeyPrivate,
-          this.walletLocalData.moneroSpendKeyPublic,
-          this.walletInfo.keys.moneroSpendKeyPrivate
-        )
-        if (this.walletLocalData.blockHeight !== parsedAddrInfo.blockchain_height) {
-          this.walletLocalData.blockHeight = parsedAddrInfo.blockchain_height // Convert to decimal
-          this.walletLocalDataDirty = true
-          this.edgeTxLibCallbacks.onBlockHeightChanged(this.walletLocalData.blockHeight)
-        }
+      const addrResult = await this.myMoneroApi.getAddressInfo(params)
 
-        const nativeBalance = bns.sub(parsedAddrInfo.total_received_String, parsedAddrInfo.total_sent_String)
+      if (this.walletLocalData.blockHeight !== addrResult.blockHeight) {
+        this.walletLocalData.blockHeight = addrResult.blockHeight // Convert to decimal
+        this.walletLocalDataDirty = true
+        this.edgeTxLibCallbacks.onBlockHeightChanged(this.walletLocalData.blockHeight)
+      }
 
-        if (this.walletLocalData.totalBalances.XMR !== nativeBalance) {
-          this.walletLocalData.totalBalances.XMR = nativeBalance
-          this.edgeTxLibCallbacks.onBalanceChanged('XMR', nativeBalance)
-        }
+      const nativeBalance = bns.sub(addrResult.totalReceived, addrResult.totalSent)
 
-        this.walletLocalData.lockedXmrBalance = parsedAddrInfo.locked_balance_String
+      if (this.walletLocalData.totalBalances.XMR !== nativeBalance) {
```

Comment:
Add locked_balance_String to MyMoneroApi and copy into `walletLocalData` per @thehobbit85